sdcm/commit_log_check_thread.py

```diff
@@ -0,0 +1,192 @@
+import traceback
+import logging
+import time
+import threading
+from dataclasses import dataclass
+from distutils.util import strtobool
+from sdcm.sct_events.database import CommitLogCheckErrorEvent
+
+
+@dataclass
+class CommitlogConfigParams:
+    def __init__(self, db_cluster, ):
+        log = logging.getLogger(self.__class__.__name__)
+
+        with db_cluster.cql_connection_patient(
+                node=db_cluster.nodes[0],
+                connect_timeout=300,) as session:
+            self.total_space_in_mb = int(session.execute(
+                "SELECT value FROM system.config WHERE name='commitlog_total_space_in_mb'").one().value)
+            self.use_hard_size_limit = bool(strtobool(session.execute(
+                "SELECT value FROM system.config WHERE name='commitlog_use_hard_size_limit'").one().value))
+            self.segment_size_in_mb = int(session.execute(
+                "SELECT value FROM system.config WHERE name='commitlog_segment_size_in_mb'").one().value)
+
+            if self.total_space_in_mb == -1:
+                self.total_space_in_mb = "scylla_memory_total_memory"
+            else:
+                self.total_space_in_mb = self.total_space_in_mb * 1048576
+
+            log.warning("commitlog_total_space_in_mb")
+            log.warning(self.total_space_in_mb)
+
+            log.warning("commitlog_use_hard_size_limit")
+            log.warning(self.use_hard_size_limit)
+
+            log.warning("commitlog_segment_size_in_mb")
+            log.warning(self.segment_size_in_mb)
+
+    total_space_in_mb: int | str
+    use_hard_size_limit: bool
+    segment_size_in_mb: int
+
+
+@dataclass
+class PrometheusQueries:
+    def __init__(self, commitlog_params: CommitlogConfigParams):
+        self.commit_log_directory_query = self.commit_log_directory_query.format(
+            commitlog_total_space_in_mb=commitlog_params.total_space_in_mb,
+            commitlog_segment_size_in_mb=commitlog_params.segment_size_in_mb
+        )
+
+        self.free_segments_query = self.free_segments_query.format(
+            commitlog_total_space_in_mb=commitlog_params.total_space_in_mb,
+            commitlog_segment_size_in_mb=commitlog_params.segment_size_in_mb
+        )
+    # Prometheus queries with detailed description below
+    commit_log_directory_query = ("scylla_commitlog_disk_total_bytes>({commitlog_total_space_in_mb}"
```

Comment:
because returning  any value means an error - data will be returned per shard:

```
{'metric': {'__name__': 'scylla_commitlog_disk_total_bytes', 'instance': '10.4.3.68', 'job': 'manager_agent', 'shard': '0'}, 'values': [[1698765005.833, '8757706752'], [1698765006.833, '8757706752']..............
```